dbms/src/Flash/tests/gtest_compute_server.cpp

```diff
@@ -147,11 +147,16 @@ class ComputeServerRunner : public DB::tests::MPPTaskTestUtils
             BlockInputStreamPtr stream;
             try
             {
-                auto tasks = prepareMPPTasks(
```

Comment:
how about `prepareMPPTasks([&] () { return context.xxx }, properties)`?
and then `DAGRequestBuilder` will under lock

Reply:
Maybe `MockDAGRequestContext::index` should be atomic_size_t?

Reply:
Sounds great, I'll change it.

Reply:
Maybe `thread_local size_t index`? but seems that mac support `thread_local` :(